.evergreen/compile-unix.sh

```diff
@@ -85,6 +85,7 @@ DEBUG_AND_RELEASE_FLAGS="\
    -DCMAKE_PREFIX_PATH=$INSTALL_DIR \
    -DCMAKE_INSTALL_PREFIX=$INSTALL_DIR \
    -DENABLE_SHM_COUNTERS=$ENABLE_SHM_COUNTERS \
+   -DBUILD_VERSION=1.16.3-pre \
```

Comment:
This is to fix variants with older versions of git. Example failure:
https://evergreen.mongodb.com/task/mongo_c_driver_r1.16_gcc48rhel_debug_compile_compression_zlib_7b771dfa58c60a9a486f20e114d1a2cefc0e1667_20_05_12_16_00_51